packages/core/src/notifications/controller.ts

```diff
@@ -121,8 +122,25 @@ export class NotificationsController {
             getLogger('notifications').verbose('No new notifications for category: %s', category)
             return
         }
-
-        getLogger('notifications').verbose('ETAG has changed for notifications category: %s', category)
+        // Parse the notifications
+        const newPayload = JSON.parse(response.content)
+        const newNotifications = newPayload.notifications ?? []
+
+        // Get the current notifications
+        const currentNotifications = this.state[category].payload?.notifications ?? []
+        const currentNotificationIds = new Set(currentNotifications.map((n: any) => n.id))
+
+        // Compare and find if there's any notifications newly added
+        const addedNotifications = newNotifications.filter((n: any) => !currentNotificationIds.has(n.id))
+
+        if (addedNotifications.length > 0) {
+            getLogger('notifications').info(
+                'New notifications received for category %s, ids: %s',
+                category,
+                addedNotifications.map((n: any) => n.id).join(', ')
+            )
+            await this.onReceiveNotifications(addedNotifications)
```

Comment:
We will only want to run this for notifications that pass the rule engine, so we will need to do that outside this function, e.g. in `displayNotifications`

Reply:
Create a new state `newlyReceived` to handle this